game/world/managers/objects/creature/CreatureManager.py

```diff
@@ -249,6 +251,10 @@ def set_virtual_item(self, slot, item_entry):
             if slot == 1:
                 self.wearing_offhand_weapon = (item_template.inventory_type == InventoryTypes.WEAPON or
                                                item_template.inventory_type == InventoryTypes.WEAPONOFFHAND)
+            # Ranged
+            if slot == 2:
+                self.wearing_offhand_weapon = (item_template.inventory_type == InventoryTypes.RANGED or
```

Comment:
I think this should be `self.wearing_ranged_weapon` :P

Reply:
...Indeed :D